method.rb

```diff
@@ -0,0 +1,190 @@
+module Enumerable
+  def my_each
+    return 'no block given' unless block_given?
+
+    i = 0
+    while i < size
+      yield self[i]
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    return 'no block given' unless block_given?
+
+    i = 0
+    while i < size
+      yield self[i], i
+      i += 1
+    end
+    self
+  end
+
+  def my_select
+    return 'no block given' unless block_given?
+
+    output_array = []
+    i = 0
+    while i < size
+      output_array << self[i] if (yield i) == true
+      i += 1
+    end
+    output_array
+  end
+
+  def my_all?
+    return 'no block given' unless block_given?
+
+    status_cumulator = true
+    i = 0
+    loop do
+      status_cumulator &&= yield i
+      i += 1
+      break if i == size || status_cumulator == false
+    end
+    status_cumulator
+  end
+
+  def my_any?
+    return 'no block given' unless block_given?
+
+    status_cumulator = false
+    i = 0
+    loop do
+      status_cumulator ||= yield i
+      i += 1
+      break if i == size || status_cumulator == true
+    end
+    status_cumulator
+  end
+
+  def my_none?
+    return 'no block given' unless block_given?
+
+    status_cumulator = true
+    i = 0
+    loop do
+      status_cumulator &&= yield i
+      i += 1
+      break if i == size || status_cumulator == false
+    end
+    !status_cumulator
+  end
+
+  def my_count
+    return size unless block_given?
+
+    counter = 0
+    i = 0
+    while i < size
+      counter += 1 if yield self[i]
+      i += 1
+    end
+    counter
+  end
+
+  def my_inject
+    return 'no block given' unless block_given?
```

Comment:
* [ ] There should be a working `my_inject`  identical and returns the same thing as ruby's [`inject`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-inject). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_inject.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

1. ```p (5..10).my_inject { |sum, n| sum + n }``` should return  ```45```
2. ```p (5..10).my_inject(1) { |product, n| product * n }``` should return ```151200```